modules/tinymce/src/core/main/ts/keyboard/PreventNoneditableInput.ts

```diff
@@ -3,11 +3,22 @@ import { Arr } from '@ephox/katamari';
 import Editor from '../api/Editor';
 import * as EditableRange from '../selection/EditableRange';
 
+const isValidContainer = (root: Element, container: Node) => root === container || root.contains(container);
+
+const isInEditableRange = (editor: Editor, range: StaticRange) => {
+  // If the range is not in the body then it's in a shadow root and we should allow that more details in: TINY-11446
+  if (!isValidContainer(editor.getBody(), range.startContainer) || !isValidContainer(editor.getBody(), range.endContainer)) {
```

Comment:
You might be able to use `SugarBody.inBody()` but it will traverse to the shadow host if it can 🤔